### PR TITLE
MM-11223: Add backup, clear and restore logic for postsInChannel

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
Add a way to clear posts of a channel for new unread/permalink views

#### Ticket Link
[MM-11233](https://mattermost.atlassian.net/browse/MM-11223)

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit tests passed
- [x] Ran `make flow` to ensure type checking passed
- [x] Added or updated unit tests (required for all new features)
- [ ] All new/modified APIs include changes to the drivers

### CODE REVIEW

File: src/reducers/entities/posts.js [hunk 1]
```diff
@@ -359,6 +359,31 @@ function handleRemovePost(posts = {}, postsInChannel = {}, postsInThread = {}, a
     return {posts: nextPosts, postsInChannel: nextPostsForChannel, postsInThread: nextPostsForThread || postsInThread};
 }
 
+function clearChannelPosts(posts, postsInChannel, postsInThread, action) {
```
Comment (saturninoabril) [inline src/reducers/entities/posts.js]: `posts` and `postsInChannel` seems to have nothing to do here.  Should it still need to pass here?

File: src/reducers/entities/posts.js [hunk 2]
```diff
@@ -359,6 +359,31 @@ function handleRemovePost(posts = {}, postsInChannel = {}, postsInThread = {}, a
...
+
+    Reflect.deleteProperty(nextPostsInChannel, action.data.channelId);
+    return {
+        posts,
+        postsInChannel: nextPostsInChannel,
+        postsInThread,
+    };
+}
+
+function addPostIdsToChannel(posts, postsInChannel, postsInThread, action) {
```
Comment (saturninoabril) [inline src/reducers/entities/posts.js]: same here

File: src/reducers/entities/posts.js (hunk 1 above)
Comment (sudheerDev) [reply src/reducers/entities/posts.js]: Yup, They are not needed. kinda went with the existing pattern. Have a look at the changes code and let me know if this is any better.

Comment (saturninoabril) [reply src/reducers/entities/posts.js]: 👍 I'm good with that.

**Comment by sudheerDev** [comment]
@hmhealey Yes, I was taking a copy of postsInChannel so, I can immediately clear the present postId's and use it in case of request failures. 
It makes sense. Thanks. I will change the variable to `restoreBackedUpPostIds`

**Comment by sudheerDev** [comment]
Need to migrate some code from webapp.

**Comment by hmhealey** [comment]
@sudheerDev Ready to merge once the build failure is fixed

**Pull request merged**